bigquery_etl/glam/models.py

```diff
@@ -177,3 +177,28 @@ def scalar_percentiles(**kwargs):
         """,
         **kwargs,
     )
+
+
+def user_counts(**kwargs):
+    """Variables for user counts."""
+    attributes = ["ping_type", "os", "app_version", "app_build_id", "channel"]
+
+    # If the set of attributes grows, the max_combinations can be set only
+    # compute a shallow set for less query complexity
+    max_combinations = len(attributes)
+    attribute_combinations = []
+    for subset_size in reversed(range(max_combinations + 1)):
+        for grouping in combinations(attributes, subset_size):
+            # channel and app_version are required in the GLAM frontend
+            if "channel" not in grouping or "app_version" not in grouping:
+                continue
+            select_expr = []
+            for attribute in attributes:
+                select_expr.append((attribute, attribute in grouping))
+            attribute_combinations.append(select_expr)
```

Comment:
nit: why not factor out this snippet of code for combinations into a separate function? Seems redundant in 2 other locations. Though while we're here, I can't remember why we don't have the check for `channel` and `app_version` to be in `grouping` for `scalar_percentiles?`

Reply:
I'll factor this out -- these combinations should be the same for all of the queries that compute the data cube? I think `scalar_percentiles` was probably an oversight.